src/azure-cli/azure/cli/command_modules/cdn/custom.py

```diff
@@ -71,6 +91,304 @@ def update_endpoint(instance,
+
+
+def add_action(client, resource_group_name, profile_name, endpoint_name,
+               rule_name, action_name, cache_behavior=None, cache_duration=None,
+               header_action=None, header_name=None, header_value=None, query_string_behavior=None,
+               query_parameters=None, redirect_type=None, redirect_protocol=None, custom_hostname=None,
+               custom_path=None, custom_querystring=None, custom_fragment=None, source_pattern=None,
+               destination=None, preserve_unmatched_path=None):
+
+    endpoint = client.endpoints.get(resource_group_name, profile_name, endpoint_name)
+    policy = endpoint.delivery_policy
+    action = create_action(action_name, cache_behavior, cache_duration, header_action, header_name,
+                           header_value, query_string_behavior, query_parameters, redirect_type,
+                           redirect_protocol, custom_hostname, custom_path, custom_querystring,
+                           custom_fragment, source_pattern, destination, preserve_unmatched_path)
+    for i in range(0, len(policy.rules)):
+        if policy.rules[i].name == rule_name:
+            policy.rules[i].actions.append(action)
+
+    params = EndpointUpdateParameters(
+        delivery_policy=policy
+    )
+
+    return client.endpoints.update(resource_group_name, profile_name, endpoint_name, params)
+
+
+def remove_rule(client, resource_group_name, profile_name, endpoint_name, rule_name):
+
+    endpoint = client.endpoints.get(resource_group_name, profile_name, endpoint_name)
+    policy = endpoint.delivery_policy
+    if policy is not None:
+        for rule in policy.rules:
+            if rule.name == rule_name:
+                policy.rules.remove(rule)
+
+    params = EndpointUpdateParameters(
+        delivery_policy=policy
+    )
+
+    return client.endpoints.update(resource_group_name, profile_name, endpoint_name, params)
+
+
+def remove_condition(client, resource_group_name, profile_name, endpoint_name, rule_name, index):
+
+    endpoint = client.endpoints.get(resource_group_name, profile_name, endpoint_name)
+    policy = endpoint.delivery_policy
+    if policy is not None:
+        for i in range(0, len(policy.rules)):
+            if policy.rules[i].name == rule_name:
+                policy.rules[i].conditions.pop(int(index))
+
+    params = EndpointUpdateParameters(
+        delivery_policy=policy
+    )
+
+    return client.endpoints.update(resource_group_name, profile_name, endpoint_name, params)
+
+
+def remove_action(client, resource_group_name, profile_name, endpoint_name, rule_name, index):
+
+    endpoint = client.endpoints.get(resource_group_name, profile_name, endpoint_name)
+    policy = endpoint.delivery_policy
+    if policy is not None:
+        for i in range(0, len(policy.rules)):
+            if policy.rules[i].name == rule_name:
+                policy.rules[i].actions.pop(int(index))
```

Comment:
could you add exception and useful error message here?

Reply:
If there is nothing to remove, the command will return the latest policy to customer. I don't think we need to throw an exception in this case